sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -470,7 +495,19 @@ def test_error_on_missing_requests_for_sub_estimator(metaestimator):
             instance = cls(**kwargs)
             with pytest.raises(UnsetMetadataPassedError, match=re.escape(msg)):
                 method = getattr(instance, method_name)
-                method(X, y, **method_kwargs)
+                if "fit" not in method_name:
+                    # set request on fit
+                    set_requests(estimator, methods=["fit"], metadata_name=key)
+                    # make sure error message corresponding to `method_name`
+                    # is used for test
+                    if method_name != "score":
+                        set_requests(estimator, methods=["score"], metadata_name=key)
```

Comment:
the information on which metadata to set, should be in the dictionary where we declare the tests. This:

```
        "requests_set_together": {"fit": ["fit", "score"]},
```

then we don't need to special case here.

Reply:
Yes, we still do need that special case here, because in this case we need to deviate from how the test flows.

The test rotates through fit, predict and score and if we're at anything different than score, that we want to test if it raises an error, we need to set the score request, not the ones set together with fit.